Xamarin.Forms.Core/Device.cs

```diff
@@ -103,11 +103,45 @@ public static void BeginInvokeOnMainThread(Action action)
 			PlatformServices.BeginInvokeOnMainThread(action);
 		}
 
-		public static double GetNamedSize(NamedSize size, Element targetElement)
+		internal static IDispatcher GetDispatcher(Element element)
 		{
-			return GetNamedSize(size, targetElement.GetType());
+			Page page = GetElementPage(element);
```

Comment:
Can you make it a little easier?
`return GetElementPage(element)?.Dispatcher ?? default(IDispatcher);`

Reply:
Changed!